### ISSUE TITLE
Error when joining dungeons

### ISSUE THREAD

**Issue by CrusaderEsper** [Chalsean/XFaction#405]
Happened upon joining SM in Pandaria remix as well as a scenario. Might have to do with them being slightly different for Horde vs Alliance.
```
1x ...ns/XFaction/Core/GameData/Zone/DungeonCollection.lua:81: attempt to call method 'ObjectName' (a nil value)
[string "@XFaction/Core/GameData/Zone/DungeonCollection.lua"]:81: in function `catchBlock'
[string "@XFaction/Core/Function/Exception/try-catch-finally.lua"]:30: in function `catch'
[string "@XFaction/Core/GameData/Zone/DungeonCollection.lua"]:80: in function `_Function'
[string "@XFaction/Core/System/Wrapper/EventCollection.lua"]:40: in function <...Ons/XFaction/Core/System/Wrapper/EventCollection.lua:22>

Locals:
err = "...ns/XFaction/Core/GameData/Zone/DungeonCollection.lua:67: attempt to call method 'ObjectName' (a nil value)"
(*temporary) = <function> defined @XFaction/Core/Function/Logger/Logger.lua:36
(*temporary) = <table> {
 GetLinkKey = <function> defined @XFaction/Core/System/Network/Link.lua:52
 SetupMenus = <function> defined @XFaction/Core/UI/Configuration/General.lua:112
 Media = <table> {
 }
 Frames = <table> {
 }
 Config = <table> {
 }
 Function = <table> {
 }
 Factories = <table> {
 }
 Settings = <table> {
 }
 ConfigDB = <table> {
 }
 Cache = <table> {
 }
 CoreInit = <function> defined @XFaction/Core/Initialize.lua:6
 DecodeBNetMessage = <function> defined @XFaction/Core/System/Network/Decode.lua:129
 EncodeChatMessage = <function> defined @XFaction/Core/System/Network/Encode.lua:73
 Initialized = true
 Timers = <table> {
 }
 Lib = <table> {
 }
 Hooks = <table> {
 }
 Name = "XFaction"
 Targets = <table> {
 }
 Events = <table> {
 }
 Links = <table> {
 }
 Friends = <table> {
 }
 AddonName = "XFaction"
 Object = <table> {
 }
 DeserializeUnitData = <function> defined @XFaction/Core/System/Network/Decode.lua:50
 Start = 1716825393
 Enum = <table> {
 }
 Error = <function> defined @XFaction/Core/Function/Logger/Logger.lua:28
 SingleLine = <function> defined @XFaction/Core/Function/Logger/Logger.lua:73
 SetupRealms = <function> defined @XFaction/Core/UI/Configuration/General.lua:48
 Defaults = <table> {
 }
 Stop = <function> defined @XFaction/Core/Initialize.lua:90
 Warn = <function> defined @XFaction/Core/Function/Logger/Logger.lua:36
 ConfigInitialize = <function> defined @XFaction/Core/UI/Configuration/General.lua:918
 Class = <table> {
 }
 SortGuildColumns = <function> defined @XFaction/Core/UI/Configuration/DataText.lua:21
 Install = <function> defined @XFaction/Core/UI/Configuration/General.lua:1014
 InitProfile = <function> defined @XFaction/Core/UI/Configuration/General.lua:1000
 WoW = <table> {
 }
 DecodeChatMessage = <function> defined @XFaction/Core/System/Network/Decode.lua:124
 EncodeBNetMessage = <function> defined @XFaction/Core/System/Network/Encode.lua:81
 RGBPercToHex = <function> defined @XFaction/Core/Function/Font/RGBToHex.lua:4
 Handlers = <table> {
 }
 SerializeUnitData = <function> defined @XFaction/Core/System/Network/Encode.lua:38
 DoubleLine = <function> defined @XFaction/Core/Function/Logger/Logger.lua:77
 ObjectsEquals = <function> defined @XFaction/Core/System/Abstract/Object.lua:142
 DataDumper = <function> defined @XFaction/Core/Function/Logger/Logger.lua:69
 Options = <table> {
 }
 Addons = <table> {
 }
 DataText = <table> {
 }
 Verbosity = 0
 Icons = <table> {
 }
 Trace = <function> defined @XFaction/Core/Function/Logger/Logger.lua:52
 Debug = <function> defined @XFaction/Core/Function/Logger/Logger.lua:48
 Player = <table> {
 }
 Title = "|cffFF4700X|r|cff33ccffFaction|r"
 Info = <function> defined @XFaction/Core/Function/Logger/Logger.lua:44
 Nodes = <table> {
 }
 ChangeLog = <table> {
 }
 Version = <table> {
 }
 Mailbox = <table> {
 }
}
(*temporary) = nil
(*temporary) = Frame {
 0 = <userdata>
}
(*temporary) = "attempt to call method 'ObjectName' (a nil value)"
XF = <table> {
 GetLinkKey = <function> defined @XFaction/Core/System/Network/Link.lua:52
 SetupMenus = <function> defined @XFaction/Core/UI/Configuration/General.lua:112
 Media = <table> {
 }
 Frames = <table> {
 }
 Config = <table> {
 }
 Function = <table> {
 }
 Factories = <table> {
 }
 Settings = <table> {
 }
 ConfigDB = <table> {
 }
 Cache = <table> {
 }
 CoreInit = <function> defined @XFaction/Core/Initialize.lua:6
 DecodeBNetMessage = <fu
```

**Comment by Chalsean** [comment]
Hmm, that's the logic for mythic keys.  Not sure why it's firing for panda land stuff.  I'll get a fix out, thx for the constant feedback

**Comment by Chalsean** [comment]
Fixed in 4.12.4
